samples/kvssink_intermittent_sample.cpp

```diff
@@ -0,0 +1,293 @@
+std::condition_variable cv;
+
+typedef enum _StreamSource {
+    TEST_SOURCE,
+    DEVICE_SOURCE
+} StreamSource;
+
+typedef struct _CustomData {
+    _CustomData() :
+            main_loop(NULL),
+            pipeline(NULL) {}
+
+    GMainLoop *main_loop;
+    GstElement *pipeline;
+} CustomData;
+
+void sigint_handler(int sigint){
+    LOG_DEBUG("SIGINT received.  Exiting...");
+    terminated = TRUE;
+    cv.notify_all();
+    if(main_loop != NULL){
+        g_main_loop_quit(main_loop);
+    }
+}
+
+static gboolean
+bus_call (GstBus *bus, GstMessage *msg, gpointer data)
+{
+    GMainLoop *loop = (GMainLoop *) ((CustomData *)data)->main_loop;
+    GstElement *pipeline = (GstElement *) ((CustomData *)data)->pipeline;
+
+    switch (GST_MESSAGE_TYPE (msg)) {
+        case GST_MESSAGE_EOS: {
+            LOG_DEBUG("[KVS sample] Received EOS message");
+            cv.notify_all();
+            break;
+        }
+
+        case GST_MESSAGE_ERROR: {
+            gchar  *debug;
+            GError *error;
+
+            gst_message_parse_error (msg, &error, &debug);
+            g_free (debug);
+
+            LOG_ERROR("[KVS sample] GStreamer error: " << error->message);
+            g_error_free (error);
+
+            g_main_loop_quit (loop);
+            break;
+        }
+
+        default: {
+            break;
+        }
+    }
+
+    return TRUE;
+}
+
+void determine_aws_credentials(GstElement *kvssink, char* streamName) {
+    char const *iot_credential_endpoint;
+    char const *cert_path;
+    char const *private_key_path;
+    char const *role_alias;
+    char const *ca_cert_path;
+    char const *credential_path;
+    if (nullptr != (iot_credential_endpoint = getenv("IOT_GET_CREDENTIAL_ENDPOINT")) &&
+        nullptr != (cert_path = getenv("CERT_PATH")) &&
+        nullptr != (private_key_path = getenv("PRIVATE_KEY_PATH")) &&
+        nullptr != (role_alias = getenv("ROLE_ALIAS")) &&
+        nullptr != (ca_cert_path = getenv("CA_CERT_PATH"))) {
+    // Set the IoT Credentials if provided in envvar.
+    GstStructure *iot_credentials =  gst_structure_new(
+            "iot-certificate",
+            "iot-thing-name", G_TYPE_STRING, streamName, 
+            "endpoint", G_TYPE_STRING, iot_credential_endpoint,
+            "cert-path", G_TYPE_STRING, cert_path,
+            "key-path", G_TYPE_STRING, private_key_path,
+            "ca-path", G_TYPE_STRING, ca_cert_path,
+            "role-aliases", G_TYPE_STRING, role_alias, NULL);
+    
+    g_object_set(G_OBJECT (kvssink), "iot-certificate", iot_credentials, NULL);
+        gst_structure_free(iot_credentials);
+    // kvssink will search for long term credentials in envvar automatically so no need to include here
+    // if no long credentials or IoT credentials provided will look for credential file as last resort.
+    } else if(nullptr != (credential_path = getenv("AWS_CREDENTIAL_PATH"))){
+        g_object_set(G_OBJECT (kvssink), "credential-path", credential_path, NULL);
+    }
```

Comment:
Can we add some debug logs in each branch to say which credentials path was chosen.